backend/data.go

```diff
@@ -37,11 +41,45 @@ func (fn QueryDataHandlerFunc) QueryData(ctx context.Context, req *QueryDataRequ
 // QueryDataRequest contains a single request which contains multiple queries.
 // It is the input type for a QueryData call.
 type QueryDataRequest struct {
+	ForwardHTTPHeaders
 	PluginContext PluginContext
 	Headers       map[string]string
 	Queries       []DataQuery
 }
 
+func (req *QueryDataRequest) SetHTTPHeader(key, value string) {
+	if req.Headers == nil {
+		req.Headers = map[string]string{}
+	}
+
+	req.Headers[fmt.Sprintf("http_%s", key)] = value
```

Comment:
Reason to this would be to allow the [environment info](https://github.com/grafana/grafana-plugin-sdk-go/blob/c2f16c63bedb935aee63623538b11ba4414de179/proto/backend.proto#L122-L123) to hold other things than HTTP headers.